OpenSim/Simulation/Model/Muscle.cpp

```diff
@@ -649,6 +649,46 @@ void Muscle::updateGeometry(const SimTK::State& s)
     updGeometryPath().updateGeometry(s);
 }
 
+void Muscle::copyPropertiesFromObject(const OpenSim::Muscle& fromObject)
+{
+    // Cycle thru properties, find name and check if this object has similarly named Property
+    // if so, delegate to Property assignment to copy values. 
+    // Special cases: Lists, Objects
+    //std::cout << (const_cast<OpenSim::Muscle&>(fromObject)).dump() << std::endl;
+
+    const int numProps = getNumProperties();
+    bool debug = false;
+    for (int px = 0; px < numProps; ++px) {
+        AbstractProperty& fromProp = updPropertyByIndex(px);
+        const std::string& pName = fromProp.getName();
+        if (hasProperty(pName)) {
+            AbstractProperty& myProp = updPropertyByName(pName); // Get writable reference to my property
+            if (myProp.isSamePropertyClass(fromProp) && !fromProp.getValueIsDefault()) {
+                if (fromProp.isOneObjectProperty()) { // Either recur or "clone the object
+                    // Expands to Property<GeometryPath>::updAs(fromProp).updValue() and works perfectly
```

Comment:
These are the two ways to get GeometryPath from Muscle one using Properties, the other using custom/macro-based methods. @chrisdembia @aseth1 I maybe missing something obvious here, if not I'm open to suggestions. Thanks